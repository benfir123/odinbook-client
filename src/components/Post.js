import { Avatar, Divider, Typography, Button, Paper } from "@mui/material";
import ThumbUpIcon from "@mui/icons-material/ThumbUp";
import CommentIcon from "@mui/icons-material/Comment";
import CommentContainer from "./CommentContainer";
import NewCommentForm from "./NewCommentForm";

const Post = ({
  post,
  user,
  handlePostLike,
  handleCommentSubmit,
  handleCommentLike,
}) => {
  return (
    <Paper
      sx={{
        p: 2,
        display: "flex",
        flexDirection: "column",
        mb: 2,
      }}
    >
      <div style={{ display: "flex", gap: 15 }}>
        <Avatar alt={post.author.full_name} src={post.author.profile_pic_url} />
        <div>
          <Typography variant="body1">{post.author.full_name}</Typography>
          <Typography variant="body2">{post.added_formatted}</Typography>
        </div>
      </div>
      <div>
        <Typography variant="body1">{post.text}</Typography>
      </div>
      <div>
        <img src="test" alt="" />
      </div>
      <Typography variant="body2">
        {post.likes.length === 1 ? "1 like" : `${post.likes.length} likes`}
      </Typography>
      <Divider />
      <div style={{ margin: 16, display: "flex", gap: 22 }}>
        <Button
          variant="outlined"
          startIcon={<ThumbUpIcon />}
          sx={{
            width: "calc(50% - 20px)",
            color: !post.likes.includes(user.id) ? "gray" : "",
            border: !post.likes.includes(user.id) ? "1px lightgray solid" : "",
          }}
          onClick={() => handlePostLike(post.id)}
        >
          Like
        </Button>
        <Button
          variant="outlined"
          startIcon={<CommentIcon />}
          sx={{
            width: "calc(50%)",
            color: "gray",
            border: "1px lightgray solid",
          }}
        >
          Comment
        </Button>
      </div>
      <Divider />
      <CommentContainer
        comments={post.comments}
        handleCommentLike={handleCommentLike}
        user={user}
      />
      <NewCommentForm
        user={user}
        handleCommentSubmit={handleCommentSubmit}
        post={post}
      />
    </Paper>
  );
};

export default Post;
