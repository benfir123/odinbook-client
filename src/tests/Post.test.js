import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Post from "../components/Post";

describe("Post component", () => {
  it("renders the correct elements", () => {
    const { container } = render(
      <Post
        user={{
          id: 1,
        }}
        post={{
          author: {
            full_name: "Ben Chanapai",
            profile_pic_url: "",
          },
          text: "Test post!",
          added_formatted: "29 Sep 2021",
          likes: [],
          comments: [],
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
