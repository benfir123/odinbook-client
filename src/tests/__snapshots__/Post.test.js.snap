// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component renders the correct elements 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-z6nvz2-MuiPaper-root"
  >
    <div
      style="display: flex; gap: 15px;"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Ben Chanapai
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          29 Sep 2021
        </p>
      </div>
    </div>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Test post!
      </p>
    </div>
    <div>
      <img
        alt=""
        src="test"
      />
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
    >
      0 likes
    </p>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      style="margin: 16px; display: flex; gap: 22px;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-cpf96q-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ThumbUpIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2z"
            />
          </svg>
        </span>
        Like
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-108zjy5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CommentIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
            />
          </svg>
        </span>
        Comment
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      style="display: flex; gap: 15px; align-items: center; margin-top: 14px;"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="Write your comment..."
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
